### ISSUE TITLE
dart2js strong: implement extractTypeArguments

### ISSUE THREAD

**Issue by sigmundch** [dart-lang/sdk#33053]
More history in #31371

**Comment by sigmundch** [comment]
The idea is to implement it as an extension method based on the calls we observe to extractTypeArguments. For example, if we see:

```dart
extractTypeArguments<Iterable>(a, f);
```

we generate:
```js
  Iterable.prototype: {
      extractTypeArguments$1: function (callback) {
        callback.call$1$0(this.$ti);
      },
```

**Comment by rakudrama** [comment]
Each class that implements `Iterable` does so in a slightly different way.
Consider `"hi".codeUnits`, `CodeUnits` implements `Iterable<int>` by extending `UnmodifiableListBase<int>`.

We already have mechanisms in place to handle `o is Iterable<T>`. This has a specialized call for when the interface raw type is known and the parameterization is present and (possibly) unknown.
This specialized function could be cloned and made to call the callback with the types instead of doing the type parameter subtype check.

To ensure the right type metadata and methods are implemented, we should in the impacts system, model `extractTypeArguments<Interface>` as `o is Interface<T>`.

**Comment by sigmundch** [comment]
@rakudrama landed a fix in ef6e19a83a77135033b464d65a0f08075bc7bcee
